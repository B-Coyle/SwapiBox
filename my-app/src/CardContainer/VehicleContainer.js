import React, { Component } from "react";
import Card from "../Card/Card";
import { fetchMaster } from "../ApiCall/apiCalls.js";
import './Container.scss'
const uuidv4 = require("uuid/v4");

class VehicleContainer extends Component {
  constructor() {
    super();
    this.state = {
      vehicles: [],
      error: "",
      isLoading: false
    };
  }

  componentDidMount() {
    this.setState({ 
      isLoading: true 
    });

    this.setVehicles();
  }

  setVehicles = () => {
    fetchMaster("vehicles")
      .then(data => this.formatVehicle(data.results))
      .catch(error => this.setState({ error }));
  };

  formatVehicle = vehicles => {
    let updatedVehicles = vehicles.map(vehicle => {
      let { name, model, passengers } = vehicle;
      let vehicleClass = vehicle.vehicle_class;
      console.log(vehicleClass)
      return { name, model, vehicleClass, passengers, key: uuidv4() };
    });
    this.setState({ 
      vehicles: updatedVehicles, 
      isLoading: false 
    });
  };

  render() {
    const vehiclesCards = this.state.vehicles.map(vehicle => (
      <Card key={vehicle.key} data={vehicle} category="vehicle" />
    ));

    const loadingMessage = (
      <div className="loadingMessage">
        <p>Information loading. Please Hold...</p>
      </div>
    );
    return (
      <section className="cardContainer vehiclesContainer">
        {this.state.isLoading ? loadingMessage : vehiclesCards}
      </section>
    );
  }
}

export default VehicleContainer;