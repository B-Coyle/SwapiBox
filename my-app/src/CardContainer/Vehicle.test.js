import React from 'react';
import VehicleContainer from '../CardContainer/VehicleContainer.js';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';


describe('VehicleContainer', () => {
    let wrapper; 
    const defaultState ={
            vehicles: [],
            error: "",
            isLoading: false
          }
    beforeEach(()=> {
    wrapper = shallow(<VehicleContainer/>);
    });

    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
        });
    
    it("should have default state", () => {
        expect(wrapper.state()).toEqual(defaultState);
    });

})