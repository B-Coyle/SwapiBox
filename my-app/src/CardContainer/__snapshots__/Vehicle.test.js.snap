// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleContainer should match the snapshot with all data passed in 1`] = `
<section
  className="cardContainer vehiclesContainer"
>
  <div
    className="loadingMessage"
  >
    <p>
      Information loading. Please Hold...
    </p>
  </div>
</section>
`;
